### ISSUE TITLE
Weird @extend behavior

### ISSUE THREAD

**Issue by Kenneth-KT** [leafo/scssphp#67]
When I test the compiler with this input, i found that the behavior of @extend is very weird

``` SCSS
.nav-bar {
    background: #eee;
    > .item {
        margin: 0 10px;
    }
}


header ul {
    @extend .nav-bar;
    > li {
        @extend .item;
    }
}
```

It gives out the following output, notice the repeated "header ul >" in the css selector (pay attention on the comma)

``` CSS
.nav-bar, header ul {
  background: #eee; }
  .nav-bar > .item, header ul > .item, header ul > header ul > li, header ul > header ul > li, .nav-bar > header ul > li, header ul > .nav-bar > li {
    margin: 0 10px; }
```

While I test the same input in the official compiler http://sass-lang.com/try.html, it gives out the correct output

``` CSS
.nav-bar, header ul {
  background: #eee; }
  .nav-bar > .item, header ul > .item, header ul.nav-bar > li, header ul > li {
    margin: 0 10px; }
```

**Comment by robocoder** [comment]
The behavior in sass 3.4.13 is a bit different:
```
.nav-bar, header ul {
  background: #eee; }
  .nav-bar > .item, header ul > .item, header ul > li {
    margin: 0 10px; }
```

**Comment by robocoder** [comment]
Note to self: narrowed it down to the call to flattenSelectors()

**Comment by niaccurshi** [comment]
This hasn't been touched for quite a while, but just a note that the problem resides in matchExtends(), potentially also mergeDirectRelationships()

**Comment by Cerdic** [comment]
Fixed with #681
